Graded/Heap/Soundness.agda

```diff
@@ -4,20 +4,24 @@
 
 open import Graded.Modality
 open import Graded.Usage.Restrictions
+open import Graded.Usage.Restrictions.Natrec
 open import Definition.Typed.Restrictions
 import Graded.Heap.Bisimilarity
 open import Tools.Sum
 
 module Graded.Heap.Soundness
   {a} {M : Set a} {𝕄 : Modality M}
-  {UR : Usage-restrictions 𝕄}
+  (UR : Usage-restrictions 𝕄)
   (TR : Type-restrictions 𝕄)
-  (open Graded.Heap.Bisimilarity UR TR)
-  (open Type-restrictions TR)
+  (open Usage-restrictions UR)
+  (factoring-nr :
+    ⦃ has-nr : Nr-available ⦄ →
+    Is-factoring-nr M (Natrec-mode-Has-nr 𝕄 has-nr))
+  (open Graded.Heap.Bisimilarity UR TR factoring-nr)
   (As : Assumptions)
```

Comment:
Here you have one assumption that says that if the rule with an nr function is used, then that nr function is factoring, then you have another assumption *hidden inside `Assumptions`* that says that the `Nr-not-available` rule is not used. In the interests of readability I think this telescope should be refactored. Perhaps you could have a positive assumption inside `Assumptions` that states that either `Nr-not-available-GLB` holds, or `Nr-available` holds for a factoring nr function.

The same comment applies to `Graded.Heap.Termination` as well.